pkg/controller/nodelink/nodelink_controller.go

```diff
@@ -210,6 +213,15 @@ func (r *ReconcileNodeLink) Reconcile(request reconcile.Request) (reconcile.Resu
 		return reconcile.Result{}, nil
 	}
 
+	if err := r.updateNodeRef(machine, node); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error updating nodeRef for machine %q and node %q: %v", machine.GetName(), node.GetName(), err)
+	}
+
+	if !node.DeletionTimestamp.IsZero() {
```

Comment:
Why updating NodeRef on the machine but skipping to update machine annotation on node?

Reply:
nevermind. Now i see handling of deletion timsestamp case in `updateNodeRef`.